core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -104,27 +104,30 @@ object ZkUtils {
    * Get calls that only depend on static paths
    */
   def getTopicPath(topic: String): String = {
-    ZkUtils.BrokerTopicsPath + "/" + topic
+    getZkPath(ZkUtils.BrokerTopicsPath, topic)
   }
 
   def getTopicPartitionsPath(topic: String): String = {
-    getTopicPath(topic) + "/partitions"
+    getZkPath(getTopicPath(topic), "partitions")
   }
 
   def getTopicPartitionPath(topic: String, partitionId: Int): String =
-    getTopicPartitionsPath(topic) + "/" + partitionId
+    getZkPath(getTopicPartitionsPath(topic), partitionId)
 
   def getTopicPartitionLeaderAndIsrPath(topic: String, partitionId: Int): String =
-    getTopicPartitionPath(topic, partitionId) + "/" + "state"
+    getZkPath(getTopicPartitionPath(topic, partitionId), "state")
     
   def getEntityConfigRootPath(entityType: String): String =
-    ZkUtils.EntityConfigPath + "/" + entityType
+    getZkPath(ZkUtils.EntityConfigPath, entityType)
 
   def getEntityConfigPath(entityType: String, entity: String): String =
-    getEntityConfigRootPath(entityType) + "/" + entity
+    getZkPath(getEntityConfigRootPath(entityType), entity)
 
   def getDeleteTopicPath(topic: String): String =
-    DeleteTopicsPath + "/" + topic
+    getZkPath(DeleteTopicsPath, topic)
+
+  def getZkPath(nodes: Any*): String =
```

Comment:
In my opinion:
* `nodes` should be of type `String*`
* We should be exposing higher level methods like the offset one and ideally in a way where paths are not built by the callers.